packages/boxel/addon/utils/reflection.ts

```diff
@@ -0,0 +1,20 @@
+const isGetter = (obj: unknown, name: string): boolean =>
+  !!(Object.getOwnPropertyDescriptor(obj, name) || {}).get;
+
+const isFunction = (obj: Record<string, unknown>, name: string): boolean =>
+  typeof obj[name] === 'function';
+
+const deepFunctions = (obj: Record<string, unknown>): null | string[] => {
+  if (!obj || obj === Object.prototype) return null;
+  return Object.getOwnPropertyNames(obj)
+    .filter((name) => isGetter(obj, name) || isFunction(obj, name))
+    .concat(deepFunctions(Object.getPrototypeOf(obj)) || []);
+};
+
+const distinctDeepFunctions = (obj: Record<string, unknown>) =>
+  Array.from(new Set(deepFunctions(obj)));
+
+export const getMethodNames = (obj: Record<string, unknown>): string[] =>
+  distinctDeepFunctions(obj).filter(
+    (name: string) => name !== 'constructor' && !~name.indexOf('__')
+  );
```

Comment:
you could use a generic type here, then you don't need to cast it: 
```ts
export const getMethodNames = <Obj extends Record<keyof Obj, unknown>>(
  obj: Obj
): string[] =>
```

Reply:
Yay, this is the concept I was missing!